pkg/scd/operational_intents_handler.go

```diff
@@ -16,6 +16,44 @@ import (
 	"github.com/interuss/stacktrace"
 )
 
+// subscriptionIsOnlyAttachedToOIR will check if:
+// - the subscription exists and is implicit
+// - the subscription is attached to the specified operational intent
+// - the subscription is not attached to any other operational intent
+//
+// This is to be used in contexts where an implicit subscription may need to be cleaned up: if true is returned,
+// the subscription can be safely removed after the operational intent is deleted or attached to another subscription.
+//
+// NOTE: this should eventually be pushed down to CRDB as part of the queries being executed in the callers of this method.
+//
+//	See https://github.com/interuss/dss/issues/1059 for more details
+func subscriptionIsOnlyAttachedToOIR(ctx context.Context, r repos.Repository, oirID, subscriptionID *dssmodels.ID) (bool, error) {
+	// Get the Subscription supporting the OperationalIntent, if one is defined
+	if subscriptionID != nil {
+		sub, err := r.GetSubscription(ctx, *subscriptionID)
+		if err != nil {
+			return false, stacktrace.Propagate(err, "Unable to get OperationalIntent's Subscription from repo")
+		}
+		if sub == nil {
+			return false, stacktrace.NewError("OperationalIntent's Subscription missing from repo")
+		}
+
+		if sub.ImplicitSubscription {
+			// Get the Subscription's dependent OperationalIntents
+			dependentOps, err := r.GetDependentOperationalIntents(ctx, sub.ID)
+			if err != nil {
+				return false, stacktrace.Propagate(err, "Could not find dependent OperationalIntents")
+			}
+			if len(dependentOps) == 0 {
+				return false, stacktrace.NewError("An implicit Subscription had no dependent OperationalIntents")
```

Comment:
Not sure if that's OK: when this new logic will be introduced, there will exist in the store some dangling implicit subscriptions.

Reply:
That is correct: although, I essentially kept the logic as it existed before being factored into this method.

I'd rather fix this in a separate PR, as there might be some discussion about whether or not we should opportunistically fix such inconsistencies when we encounter them vs. noisily failing so we become aware of them.